specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2022-03-01-preview/activeDirectoryConnectors.json

```diff
@@ -0,0 +1,232 @@
+      "description": "Active Directory domain details",
+      "properties": {
+        "realm": {
+          "type": "string",
+          "description": "Name (uppercase) of the Active Directory domain that this AD connector will be associated with."
+        },
+        "netbiosDomainName": {
+          "type": "string",
+          "description": "NETBIOS name of the Active Directory domain."
+        },
+        "domainControllers": {
+          "$ref": "#/definitions/ActiveDirectoryDomainControllers",
+          "description": "null",
+          "x-ms-client-flatten": false
+        }
+      },
+      "required": [
+        "realm",
+        "domainControllers"
+      ]
+    },
+    "ActiveDirectoryConnectorDNSDetails": {
+      "type": "object",
+      "description": "DNS server details",
+      "properties": {
+        "domainName": {
+          "type": "string",
+          "description": "DNS domain name for which DNS lookups should be forwarded to the Active Directory DNS servers."
+        },
+        "nameserverIPAddresses": {
+          "type": "array",
+          "description": "List of Active Directory DNS server IP addresses.",
+          "items": {
+            "type": "string",
+            "minimum": 1
+          }
+        },
+        "replicas": {
+          "type": "integer",
+          "description": "Replica count for DNS proxy service. Default value is 1.",
+          "default": 1,
+          "format": "int64"
+        },
+        "preferK8sDnsForPtrLookups": {
+          "type": "boolean",
+          "description": "Flag indicating whether to prefer Kubernetes DNS server response over AD DNS server response for IP address lookups.",
+          "default": true
+        }
+      },
+      "required": [
+        "nameserverIPAddresses"
+      ]
+    },
+    "ActiveDirectoryDomainControllers": {
+      "type": "object",
+      "description": "Details about the Active Directory domain controllers associated with this AD connector instance",
+      "properties": {
+        "primaryDomainController": {
+          "$ref": "#/definitions/ActiveDirectoryPrimaryDomainController",
+          "description": "null",
+          "x-ms-client-flatten": false
+        },
+        "secondaryDomainControllers": {
+          "$ref": "#/definitions/ActiveDirectorySecondaryDomainControllers",
+          "description": "null",
+          "x-ms-client-flatten": false
+        }
+      },
+      "required": [
+        "primaryDomainController"
+      ]
+    },
+    "ActiveDirectoryPrimaryDomainController": {
+      "type": "object",
+      "description": "Information about the Primary Domain Controller (PDC) in the AD domain.",
+      "properties": {
+        "hostname": {
+          "type": "string",
+          "description": "Fully-qualified domain name of the Primary Domain Controller (PDC) in the AD domain."
+        }
+      },
+      "required": [
+        "hostname"
+      ]
+    },
+    "ActiveDirectorySecondaryDomainControllers": {
```

Comment:
>ActiveDirectorySecondaryDomainControllers [](http://example.com/codeflow?start=5&length=41)

nit - probably you can simply reference `ActiveDirectoryDomainController` an array of objects whose value is a ref.

Reply:
Followed this suggestion. `ActiveDirectorySecondaryDomainControllers` now references the `ActiveDirectoryDomainController` object